doc/cookbook/type/create_custom_field_type.rst

```diff
@@ -0,0 +1,215 @@
+    need when rendering your field in a template. For example, in `IntegerType`_,
+    a ``precision`` variable is set and used in the template to set
+    the ``precision`` attribute on the ``input`` field.
+
+``configureOptions()``
+    This defines options for your field type that
+    can be used in ``buildType()`` and ``buildView()``. There are a lot of
+    options common to all fields (see :doc:`/reference/types/field`),
+    but you can create any others that you need here.
+
+Creating the ClientIdTransformer
+--------------------------------
+
+Because the type handel's client ID's in a special format, the type needs
+a :doc:`data_transformers` to transform the input back
+into a regular integer.
+
+.. code-block:: php
+    :linenos:
+
+    // src/Acme/Client/Search/DataTransformer/ClientIdTransformer.php
+
+    namespace Acme\Client\Search\DataTransformer;
+
+    use Rollerworks\Component\Search\DataTransformerInterface;
+    use Rollerworks\Component\Search\Exception\TransformationFailedException;
+
+    class ClientIdTransformer implements DataTransformerInterface
+    {
+        public function transform($value)
+        {
+            return sprintf('C%04d', $value);
+        }
+
+        public function reverseTransform($value)
+        {
+            if (null !== $value && !is_scalar($value)) {
+                throw new TransformationFailedException('Expected a scalar.');
+            }
+
+            return ltrim('C0');
+        }
+    }
+
+Using the Field Type
+--------------------
+
+Now the type is created, the Search system needs a way to find it.
+
+This can be done in to ways;
+
+You can choose to use your custom field type immediately, simply by creating a
+new instance of the type in one of your FieldSets:
+
+.. code-block:: php
+    :linenos:
+
+    use Acme\Client\Search\Type\ClientIdType;
+    use Rollerworks\Component\Search\Searches;
+
+    $searchFactory = new Searches::createSearchFactoryBuilder()->getSearchFactory();
+
+    $fieldset = $searchFactory->createFieldSetBuilder('clients')
+        ->add('id', new ClientIdType())
+        ->add('name', 'text')
+        ->getFieldSet()
+    ;
+
+Or the by registering the type in a ``SearchExtension``.
+
+.. tip::
+
+    Registering the type in a ``SearchExtension`` is the recommended way
+    when you want to reuse the type in multiple FieldSets or when you
+    need some additional parameters to the class constructor.
+
+.. code-block:: php
+    :linenos:
+
+    // src/Acme/Client/Search/ClientExtension.php
+
+    namespace Acme\Client\Search;
+
+    use Rollerworks\Component\Search\AbstractExtension;
+
+    class ClientExtension extends AbstractExtension
+    {
+        protected function loadTypes()
+        {
+            return array(
+                new Type\ClientIdType(),
+            );
+        }
+    }
+
+And then register it at system using the FactoryBuilder.
+
+.. code-block:: php
+
+    /* ... */
```

Comment:
Indeed, the ``/* ... */` is only used to demonstrate an alternative approach (which I actually try to avoid this way as its less readable).